include/fmt/core.h

```diff
@@ -1368,11 +1374,32 @@ template <typename Context> struct arg_mapper {
     return val;
   }
 
+  template <typename T,
+            FMT_ENABLE_IF(std::is_function<typename std::remove_pointer<
+                              remove_cvref_t<T>>::type>::value)>
+  FMT_CONSTEXPR auto map(const T&) -> unformattable_function_pointer {
+    return {};
+  }
+
+  template <typename T, FMT_ENABLE_IF(std::is_member_object_pointer<T>::value)>
+  FMT_CONSTEXPR auto map(const T&) -> unformattable_member_object_pointer {
+    return {};
+  }
+
+  template <typename T,
+            FMT_ENABLE_IF(std::is_member_function_pointer<T>::value)>
+  FMT_CONSTEXPR auto map(const T&) -> unformattable_member_function_pointer {
+    return {};
+  }
+
   // We use SFINAE instead of a const T* parameter to avoid conflicting with
   // the C array overload.
+
   template <
       typename T,
-      FMT_ENABLE_IF(std::is_convertible<const T&, const void*>::value &&
+      FMT_ENABLE_IF(!std::is_function<typename std::remove_pointer<
+                        remove_cvref_t<T>>::type>::value &&
+                    std::is_convertible<const T&, const void*>::value &&
```

Comment:
For some implementation function pointer is implicitly convertible to `void*`, so checking also here to make the resolution unambiguous. 

(So the original issue did not reproduce for MSVC)